poky rev: 753e2a0ede4449917c75353b57f13bbafe70fac8
file: https://git.yoctoproject.org/cgit/cgit.cgi/poky/tree/meta/recipes-devtools/apt/apt/0001-apt-1.2.12-Fix-musl-build.patch?id=753e2a0ede4449917c75353b57f13bbafe70fac8

rebased to 1.8 patch
pull request: https://github.com/masami256/apt/pull/3

diff --git a/apt-pkg/contrib/srvrec.h b/apt-pkg/contrib/srvrec.h
index e22b7a1c6..b1115f5ba 100644
--- a/apt-pkg/contrib/srvrec.h
+++ b/apt-pkg/contrib/srvrec.h
@@ -9,6 +9,7 @@
 #ifndef SRVREC_H
 #define SRVREC_H
 
+#include <sys/types.h>
 #include <string>
 #include <vector>
 #include <arpa/nameser.h>
diff --git a/methods/connect.cc b/methods/connect.cc
index e5d17a2f5..c8c68a3c6 100644
--- a/methods/connect.cc
+++ b/methods/connect.cc
@@ -32,6 +32,10 @@
 #include <unistd.h>
 
 // Internet stuff
+#ifndef AI_IDN
+#define AI_IDN 0x0040
+#endif
+
 #include <netdb.h>
 #include <arpa/inet.h>
 #include <netinet/in.h>
